File: dotnet_api/Repository.cs

using dotnet_api;

public interface IRepository
{
    Task ExecuteFunction(RequestInfoDataRequest data, List<string> nameList);
}

public class Repository : IRepository
{
    private readonly ILogger<RequestInfoDataRequest> _logger;
    public Repository(ILogger<RequestInfoDataRequest> logger)
    {
        this._logger = logger;
    }

    public async Task ExecuteFunction(RequestInfoDataRequest data, List<string> nameList)
    {
        var joinedName = string.Join(",", nameList);
        this._logger.LogInformation($"Executing {joinedName} with data: {data.Info.Sequence}");
        await Task.Delay(2000);
        this._logger.LogInformation($"Executed {joinedName} with data: {data.Info.Sequence}");
    }
}

File: fast_api/repository.py
import logging
import time
import json

class Repository:
    def __init__(self):
        pass

    def execute_action(self, data, name_list):
        logging.info(f'Executing {name_list} with data: {data}')
        time.sleep(2)
        return json.dumps(f'{name_list} Executed'), 200, {'ContentType': 'application/json'}
